src/drive/web/modules/upload/index.js

```diff
@@ -138,30 +176,55 @@ const uploadFile = async (client, file, dirID) => {
   return resp.data
 }
 
+export const getFilePath = async (client, file, dirID) => {
+  const resp = await client.collection('io.cozy.files').get(dirID)
+  const parentDirectory = resp.data
+  return `${parentDirectory.path}/${file.name}`
+}
+
+export const overwriteFile = async (client, file, path) => {
+  const statResp = await client.collection('io.cozy.files').statByPath(path)
+  const { id: fileId, dir_id: dirId } = statResp.data
+  const resp = await client
+    .collection('io.cozy.files')
+    .updateFile(file, { dirId, fileId })
+
+  return resp.data
+}
+
 export const addToUploadQueue = (
   files,
   dirID,
+  sharingState,
   fileUploadedCallback,
   queueCompletedCallback
 ) => async dispatch => {
   dispatch({
     type: ADD_TO_UPLOAD_QUEUE,
     files: extractFilesEntries(files)
   })
-  dispatch(processNextFile(fileUploadedCallback, queueCompletedCallback, dirID))
+  dispatch(
+    processNextFile(
+      fileUploadedCallback,
+      queueCompletedCallback,
+      dirID,
+      sharingState
+    )
+  )
 }
 
 export const purgeUploadQueue = () => ({ type: PURGE_UPLOAD_QUEUE })
 
 export const onQueueEmpty = callback => (dispatch, getState) => {
   const queue = getUploadQueue(getState())
-  const loaded = getLoaded(queue)
+  const uploaded = getUploaded(queue)
```

Comment:
why are we removing `loaded`?

Reply:
because we have to make a distinction between uploaded files (new) and updated files

Reply:
So maybe `created` and `updated` are better? Since even if we `update` we `upload`. No?

Reply:
:+1: